### ISSUE TITLE
node os cpus is POSIX-only: on Windows it is 0

### ISSUE THREAD

**Issue by userquin** [nuxt/nuxt#25360]
### Environment

```shell
------------------------------
- Operating System: Windows_NT
- Node Version:     v20.10.0
- Nuxt Version:     3.9.1
- CLI Version:      3.10.0
- Nitro Version:    2.8.1
- Package Manager:  pnpm@8.14.1
- Builder:          -
- User Config:      typescript, devtools, imports, vue, experimental
- Runtime Modules:  -
- Build Modules:    -
------------------------------
```

### Reproduction

It is a problem using `cpus()` in `packages/nuxt/src/core/nitro.ts`

NA

### Describe the bug

Check https://github.com/unjs/nitro/issues/2103

**Comment by pi0** [comment]
Considering the limitation of Windows reporting `0`, the default would have to [be 4](https://github.com/nuxt/nuxt/blob/89541f416446b983eb575dc55a06209aef3fb668/packages/nuxt/src/core/nitro.ts#L162). Is this what makes issues with OS's less than 4 cores? 

On a solution for correct default of CPU core in windows, we might use [`NUMBER_OF_PROCESSORS`](https://learn.microsoft.com/en-us/troubleshoot/windows-server/deployment/number-of-processors-environment-variable-show-incorrect-values) environment variable but probably Node.js + Nuxt fallback to `4` is safest.


---

(following up discord discussion, I think your server had more than 4 -- so probably would be nice having another issue about why you believe it can lead to memory leak in Nuxt or Nitro? -- also let's consider memory leak !== high memory usage and a reproduction or steps for it would be nice for it as well.)

**Comment by userquin** [comment]
If I'm generating pages on my local, no matter how many parallel task running, why 4GB of run? My laptop has 20 cores, so I can configure `nitro.prerender.concurrency = 80` using Nuxt formula.

Running `node ./cpus.mjs`, seems to return correct cpus using both, `cpus` and `availableParallelism` (this issue can be closed): 
```mjs
import { cpus, availableParallelism } from 'node:os'

console.log(cpus())
console.log(cpus().length)
console.log(availableParallelism())
```

<details>
<summary>ouput</summary>

```shell
node ./cpus.mjs
[
  {
    model: '12th Gen Intel(R) Core(TM) i7-12700H',
    speed: 2688,
    times: {
      user: 12417125,
      nice: 0,
      sys: 4045562,
      idle: 377224531,
      irq: 293812
    }
  },
  {
    model: '12th Gen Intel(R) Core(TM) i7-12700H',
    speed: 2688,
    times: { user: 1562437, nice: 0, sys: 342656, idle: 391782015, irq: 5656 }
  },
  {
    model: '12th Gen Intel(R) Core(TM) i7-12700H',
    speed: 2688,
    times: {
      user: 69069171,
      nice: 0,
      sys: 8240390,
      idle: 316377484,
      irq: 208265
    }
  },
  {
    model: '12th Gen Intel(R) Core(TM) i7-12700H',
    speed: 2688,
    times: {
      user: 4398921,
      nice: 0,
      sys: 770531,
      idle: 388517593,
      irq: 15468
    }
  },
  {
    model: '12th Gen Intel(R) Core(TM) i7-12700H',
    speed: 2688,
    times: {
      user: 34584265,
      nice: 0,
      sys: 5224937,
      idle: 353877843,
      irq: 116218
    }
  },
  {
    model: '12th Gen Intel(R) Core(TM) i7-12700H',
    speed: 2688,
    times: {
      user: 4572000,
      nice: 0,
      sys: 840750,
      idle: 388274265,
      irq: 15328
    }
  },
  {
    model: '12th Gen Intel(R) Core(TM) i7-12700H',
    speed: 2688,
    times: {
      user: 9071921,
      nice: 0,
      sys: 1547343,
      idle: 383067734,
      irq: 45515
    }
  },
  {
    model: '12th Gen Intel(R) Core(TM) i7-12700H',
    speed: 2688,
    times: { user: 1327406, nice: 0, sys: 242953, idle: 392116687, irq: 6109 }
  },
  {
    model: '12th Gen Intel(R) Core(TM) i7-12700H',
    speed: 2688,
    times: {
      user: 6666703,
      nice: 0,
      sys: 4533500,
      idle: 382486812,
      irq: 2737375
    }
  },
  {
    model: '12th Gen Intel(R) Core(TM) i7-12700H',
    speed: 2688,
    times: { user: 1227703, nice: 0, sys: 172171, idle: 392287171, irq: 4031 }
  },
  {
    model: '12th Gen Intel(R) Core(TM) i7-12700H',
    speed: 2688,
    times: {
      user: 5201250,
      nice: 0,
      sys: 896234,
      idle: 387589531,
      irq: 28234
    }
  },
  {
    model: '12th Gen Intel(R) Core(TM) i7-12700H',
    speed: 2688,
    times: { user: 1135687, nice: 0, sys: 124890, idle: 392426437, irq: 3625 }
  },
  {
    model: '12th Gen Intel(R) Core(TM) i7-12700H',
    speed: 2688,
    times: {
      user: 18036843,
      nice: 0,
      sys: 12133468,
      idle: 363516687,
      irq: 3857968
    }
  },
  {
    model: '12th Gen Intel(R) Core(TM) i7-12700H',
    speed: 2688,
    times: {
      user: 21591187,
      nice: 0,
      sys: 8378750,
      idle: 363717062,
      irq: 336656
    }
  },
  {
    model: '12th Gen Intel(R) Core(TM) i7-12700H',
    speed: 2688,
    times: {
      user: 19861468,
      nice: 0,
      sys: 8084250,
      idle: 365741281,
      irq: 338859
    }
  },
  {
    model: '12th Gen Intel(R) Core(TM) i7-12700H',
    speed: 2688,
    times: {
      user: 21920125,
      nice: 0,
      sys: 8089468,
      idle: 363677375,
      irq: 332515
    }
  },
  {
    model: '12th Gen Intel(R) Core(TM) i7-12700H',
    speed: 2688,
    times: {
      user: 22342781,
      nice: 0,
      sys: 8149640,
      idle: 363194562,
      irq: 376593
    }
  },
  {
    model: '12th Gen Intel(R) Core(TM) i7-12700H',
    speed: 2688,
    times: {
      user: 21645421,
      nice: 0,
      sys: 7664765,
      idle: 364376796,
      irq: 323796
    }
  },
  {
    model: '12th Gen Intel(R) Core(TM) i7-12700H',
    speed: 2688,
    times: {
      user: 20915453,
      nice: 0,
      sys: 7340125,
      idle: 365431390,
      irq: 317187
    }
  },
  {
    model: '12th Gen Intel(R) Core(TM) i7-12700H',
    speed: 2688,
    times: {
      user: 20806546,
      nice: 0,
      sys: 7480046,
      idle: 365400375,
      irq: 323953
    }
  }
]
20
20
```
</details>

**Comment by pi0** [comment]
> My laptop has 20 cores, so I can configure nitro.prerender.concurrency = 80 using Nuxt formula.

CPU is not the only limiter. Running 80 rendering tasks in parallel, means you need 80x more of the memory requirement than one page needs. Both resources need to be available enough to handle more parallel load.
